binary_search_tree/tree.py

```diff
@@ -14,44 +14,160 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    # Time Complexity: O(logn)
+    # Space Complexity: O(1)
+    def add(self, key, value):
+        new_node = TreeNode(key,value)
+        if not self.root:
+            self.root = new_node
+            return 
+        current = self.root
+        previous = None
+        while current:
+            previous = current
+            if current.key > new_node.key:
+                current = current.left
+            else:
+                current = current.right
+        if previous.key > new_node.key:
+            previous.left = new_node
+        else:
+            previous.right = new_node
 
-    # Time Complexity: 
-    # Space Complexity: 
+
+        
+    
+   
+    # Time Complexity: O(logn)
+    # Space Complexity: O(1)
     def find(self, key):
-        pass
+        current_node = self.root
+    
+        while current_node:
+            if current_node.key == key:
+                return current_node.value
+            elif current_node.key > key:
+                current_node = current_node.left
+            else:
+                current_node = current_node.right
+        return None
 
-    # Time Complexity: 
-    # Space Complexity: 
+  
+  
+    # Time Complexity: O(logn)
+    # Space Complexity: O(1)
     def inorder(self):
```

Comment:
👍 However the space and time complexities are O(n) due to the list you're building